### PR TITLE
FRONT-18-Rev

### PR DESCRIPTION
**Pull request by larasodjati**
Created a fetch wrapper for all book routes and their documentation.

### CODE REVIEW

File: src/adapters/book-adapters.ts
Comment (Manizha-khorram) [inline src/adapters/book-adapters.ts:127]: So this function will return an array not object, right? 
@returns {Promise<booksSchema[ ]>}

File: src/adapters/book-adapters.ts
Comment (Manizha-khorram) [inline src/adapters/book-adapters.ts:172]: Lara, I think this one is not returning an array. right? booksSchema[]?

File: src/adapters/book-adapters.ts
Comment (Manizha-khorram) [inline src/adapters/book-adapters.ts:240]: This one as well, I think it is returning an object.

File: src/adapters/book-adapters.ts
Comment (Manizha-khorram) [inline src/adapters/book-adapters.ts:21]: Lara, We suppose to import this fetch wrapper from fetch.ts!

File: src/adapters/book-adapters.ts
Comment (larasodjati) [reply src/adapters/book-adapters.ts:80]: You're right, I missed those queries on my code, let me add it then.

File: src/adapters/book-adapters.ts
Comment (larasodjati) [reply src/adapters/book-adapters.ts:127]: Thanks for pointing this out Manizha,  will fix it soon.

File: src/adapters/book-adapters.ts
Comment (larasodjati) [reply src/adapters/book-adapters.ts:21]: Ok,  I could import it by adding undefined on the body: object on fetch.ts

File: src/adapters/book-adapters.ts
Comment (larasodjati) [reply src/adapters/book-adapters.ts:172]: I think this one returns an array. If 1 owner has several books listed, it will return as an array right?

**Comment by Manizha-khorram** [comment]
@jpremo, Lara did commit new changes. Can you please review booksAdopter, and approve it in case it is all good from your point of view? 
Thanks

File: src/adapters/book-adapters.ts
Comment (jpremo) [inline src/adapters/book-adapters.ts:23]: Import this from fetch.ts. Like you said, you will need to change fetch.ts so that body can be undefined.

File: src/adapters/book-adapters.ts
Comment (jpremo) [inline src/adapters/book-adapters.ts:34]: Can you add on worldcatURL and ISBN here? Also add this on bookSchema

File: src/adapters/book-adapters.ts
Comment (jpremo) [inline src/adapters/book-adapters.ts:62]: This can just say createBookAdapter(bookInput) as a sufficient example. Can you change this for the other examples as well?

File: src/adapters/book-adapters.ts
Comment (jpremo) [inline src/adapters/book-adapters.ts:214]: ISBN and worldCatURL should be here too

**Review by jpremo — CHANGES_REQUESTED** [review]
Looks great! Nice work. A few very minor changes and it should be ready to merge.

File: src/adapters/book-adapters.ts
Comment (larasodjati) [reply src/adapters/book-adapters.ts:23]: @jpremo  Thank you for the feedback. Actually, I have imported fetch.ts and added body undefined to fetch.ts then when we checked the login route it gave us an error "JSON invalid/undefined....." that's why I kept this fetch file on my book-adapters, could you please advise on this issue?

File: src/adapters/book-adapters.ts
Comment (larasodjati) [reply src/adapters/book-adapters.ts:62]: Do you mean const bookInput needs to be changed to createBookAdapter(bookInput)? Other than that part in this highlight section stay the same?

File: src/adapters/book-adapters.ts
Comment (jpremo) [reply src/adapters/book-adapters.ts:23]: The function is an exact copy from fetch.ts. The login route wouldn't use the fetchAPIData defined here since it is in the auth-adapters.ts file. I tested this locally with the import and it seemed fine to me. Let me know if you can still replicate the issue and if so provide a screenshot of the entire error.

**Review by jpremo — APPROVED** [review]

**Comment by larasodjati** [comment]
@Manizha-khorram could you please approve the changes? it seems I need to ask for your approval before merging this branch. Thank you!

image

**Review by Manizha-khorram — APPROVED** [review]

**Comment by Manizha-khorram** [comment]
> @Manizha-khorram could you please approve the changes? it seems I need to ask for your approval before merging this branch. Thank you!
> 
> image

Sorry Lara, for it. I didn't know it is from my side!

**Pull request merged**
